src/axom/quest/ScatteredInterpolation.hpp

```diff
@@ -0,0 +1,402 @@
+   * \param [in] coordset The name of the coords for query points in \a query_mesh
+   * \param [in] input_mesh Root node of mesh (in blueprint format) containing the field to interpolate
+   * \param [in] input_field_name Name of field on \a input_mesh
+   * \param [in] output_field_name Name of field on \a query_mesh
+   * \param [in] INVALID_VALUE Value to use for points that are not in the \a input_mesh
+   *
+   * \pre \a input_mesh and \a query_mesh must conform to the mesh blueprint schema for point meshes
+   * \pre \a input_mesh must contain a nodal scalar field named \a input_field_name
+   * \pre \a query_mesh must contain a nodal scalar field named \a output_field_name
+   * \pre \a query_mesh must contain a nodal scalar field named \a cell_idx whose values
+   * are the index of the cell from the Delaunay triangulation containing the query points.
+   * These can be computed in the \a locatePoints() function
+   * \post Scalar field \a output_field_name on \a query_mesh will contain the interpolated
+   * values of \a input_field_name from \a input_mesh for all query points that have a valid
+   * \a cell_idx to a cell in the \a input_mesh.
+   */
+  void interpolateField(
+    conduit::Node& query_mesh,
+    const std::string& coordset,
+    conduit::Node& input_mesh,
+    const std::string& input_field_name,
+    const std::string& output_field_name,
+    const double INVALID_VALUE = std::numeric_limits<double>::quiet_NaN())
+  {
+    constexpr auto INVALID_INDEX = DelaunayTriangulation::INVALID_INDEX;
+
+    SLIC_ASSERT(this->isValidBlueprint(query_mesh));
+    SLIC_ASSERT(this->isValidBlueprint(input_mesh));
+
+    // Extract the required fields from the input and query meshes
+    auto in_fld = getField<double>(input_mesh, input_field_name);
+    auto out_fld = getField<double>(query_mesh, output_field_name);
+    auto containing_cell = getField<axom::IndexType>(query_mesh, "cell_idx");
+
+    const auto valuesPath = fmt::format("coordsets/{}/values", coordset);
+    SLIC_ASSERT(query_mesh.has_path(valuesPath));
+    auto coords = detail::InterleavedOrStridedPoints<DIM>(query_mesh[valuesPath]);
+
+    // Interpolate field at query points
+    const int npts = coords.size();
+    for(int idx = 0; idx < npts; ++idx)
+    {
+      const auto cell_id = containing_cell[idx];
+      if(cell_id == INVALID_INDEX)
+      {
+        out_fld[idx] = INVALID_VALUE;
+      }
+      else
+      {
+        double res = 0.;
+        const auto baryCoords = m_delaunay.getBaryCoords(cell_id, coords[idx]);
+        const auto verts = m_delaunay.getMeshData()->boundaryVertices(cell_id);
+        for(auto it = verts.begin(); it < verts.end(); ++it)
+        {
+          res += in_fld[*it] * baryCoords[it.index()];
+        }
+
+        out_fld[idx] = res;
+      }
+    }
+  }
+
+  /// TODO: Add a function that takes x,y,z coordinates
+  /// and returns the vertex indices and the barycentric coords
+  /// of the point w.r.t. its containing element
```

Comment:
@dtaller -- should I add this to the current PR, or save it for the next one?

Reply:
For me this function would be useful for initial testing on my end, but I will trust your judgement if you think this MR is already too long (you touched 14 files already I think). I could wait until the next one also if that makes sense.

Reply:
Thanks @dtaller -- I think I'll do a quick focused follow-up.